src/fhirtypes/common.ts

```diff
@@ -1633,3 +1633,64 @@ export function getMatchingContainedReferenceId(
     }
   }
 }
+
+export function checkForMultipleChoice(
+  fshDef: Profile | Extension | Logical | Resource | FshCodeSystem | FshValueSet | Instance,
+  fhirDef: { [key: string]: any },
+  structDef: StructureDefinition
+) {
+  checkChildrenForMultipleChoice(fshDef, fhirDef, structDef.elements[0]);
+}
+
+function checkChildrenForMultipleChoice(
+  fshDef: Profile | Extension | Logical | Resource | FshCodeSystem | FshValueSet | Instance,
+  instance: { [key: string]: any },
+  element: ElementDefinition
+) {
+  const children = element.children(true);
+  children.forEach(child => {
+    // does this child represent a choice element, such as value[x]?
+    // if so, check for choices
+    if (child.id.endsWith('[x]')) {
+      const availableChoices = child.getSlices();
```

Comment:
I'm trying to understand how this works. Does `child.getSlices()` only work if the SD used at the top-level is the same instance of the SD that was used to assign the values (e.g., it relies on choice slices having been materialized during instance assignment)? If so, that seems to me like it might be fragile (as in, it would stop working if we changed how we do assignments or if someone invoked this function w/ a fresh instance of the SD). Or am I missing something? As far as I can tell, `ElementDefinition.getSlices()` doesn't automatically expand all choice type for you -- it only works for the ones that were already expanded.

I expected you would use `element.type` to build up the list of potential value properties.